test/e2e/cassandra_test.go

```diff
@@ -0,0 +1,69 @@
+package e2e
+
+import (
+	goctx "context"
+	"github.com/instaclustr/cassandra-operator/pkg/apis"
+	operator "github.com/instaclustr/cassandra-operator/pkg/apis/cassandraoperator/v1alpha1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+)
+
+const (
+	cleanupRetryInterval = time.Second * 3
+	cleanupTimeout       = time.Second * 60
+)
+
+func TestCassandra(t *testing.T) {
+
+	cassandraList := &operator.CassandraDataCenterList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "CassandraDataCenter",
+			APIVersion: "cassandraoperator.instaclustr.com/v1alpha1",
+		},
+	}
+
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, cassandraList)
+
+	if err != nil {
+		t.Fatalf("Failed to add custom resource scheme to framework: %v", err)
+	}
+
+	t.Run("cassandra-group", func(t *testing.T) {
+		t.Run("deploy", CassandraDeploymentTest)
+	})
+}
+
+func CassandraDeploymentTest(t *testing.T) {
+
+	ctx, f := initialise(t)
+
+	cleanupOptions := &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval}
+
+	defer ctx.Cleanup()
+
+	namespace, err := ctx.GetNamespace()
+
+	if err != nil {
+		t.Fatalf("Could not get namespace: %v", err)
+	}
+
+	t.Logf("Testing Cassandra cluster will be created in namespace %s", namespace)
+
+	name := "test-dc-cassandra"
+	statefulSetName := "cassandra-test-dc-cassandra"
+
+	cassandraDC := defaultNewCassandraDataCenter(name, namespace)
+	cassandraDC.Spec.Nodes = 3
+
+	if err := f.Client.Create(goctx.TODO(), cassandraDC, cleanupOptions); err != nil {
+		t.Fatalf("Unable to create Cassandra DC object of %d node.", cassandraDC.Spec.Nodes)
```

Comment:
Is it worth including `err` in the printed output too, so that we can see why it failed?